### PR TITLE
Wizard recipe: KaHyPar-v1.2.1

### PR DESCRIPTION
**Pull request by Roger-luo**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: KaHyPar
* Version: v1.2.1

@staticfloat please review and merge.

### CODE REVIEW

**Comment by Roger-luo** [comment]
I think unfortunately kahypar is written in a non cross compilable way for windows
https://github.com/kahypar/kahypar/blob/master/kahypar/application/command_line_options.h#L25

I'm not sure how previous build script build windows successfully without install things like VS2019

File: K/KaHyPar/build_tarballs.jl [hunk 1]
```diff
@@ -3,46 +3,46 @@
...
 
 # Dependencies that must be installed before this package can be built
-dependencies = Dependency[
+dependencies = [
+    Dependency(PackageSpec(name="boost_jll", uuid="28df3c45-c428-5900-9ff8-a3135698ca75"))
 ]
 
 # Build the tarballs, and possibly a `build.jl` as well.
-build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"6.1.0")
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", preferred_gcc_version = v"10.2.0")
```
Comment (giordano) [inline K/KaHyPar/build_tarballs.jl:48]: I don't recommend at all using GCC 10.  In general use the oldest version of GCC possible, for maximum compatibility.  If GCC 6 is enough to build the library, we should stick to it

Comment (Roger-luo) [reply K/KaHyPar/build_tarballs.jl:48]: OK, I think kahypar recommends GCC 9, should we use GCC 9 then?

Comment (giordano) [reply K/KaHyPar/build_tarballs.jl:48]: Recommends or requires?  Everybody recommends the latest compiler, but we want the largest compatibility possible, which is given by the oldest GCC possible

Comment (Roger-luo) [reply K/KaHyPar/build_tarballs.jl:48]: it says "requirements" is GCC 9 or higher so I guess it is required: https://github.com/kahypar/kahypar#requirements

Comment (Roger-luo) [reply K/KaHyPar/build_tarballs.jl:48]: but it seems GCC supports C++17 since GCC 5: https://gcc.gnu.org/projects/cxx-status.html#cxx17

Comment (Roger-luo) [reply K/KaHyPar/build_tarballs.jl:48]: also should we remove the julia compat, I ran the wizard on julia 1.6, but I guess this is not required.

Comment (Roger-luo) [reply K/KaHyPar/build_tarballs.jl]: OK GCC 7 seems to be the lowest GCC version that works

**Comment by giordano** [comment]
Ok, #3115 is not going to happen anytime soon, so let's go on with this one

**Pull request merged**